File: santa.ml
let iters = int_of_string (Sys.argv.(1)) in
let l = ["Eline VDV";"Joris";"Sebastiaan";"Dirk";"Kristof";"Eline H";"Pieter"] in
let shuffle d =
  let nd = List.map (fun c -> (Random.bits (), c)) d in
  let sond = List.sort compare nd in
  List.map snd sond
in
let () = Printf.printf "The circle....." in
let print_c l = Printf.printf "%s -> %s -> " (List.nth l 0) (List.nth l 1) in
let rec loop i l =
  if i < 0 then
    ()
  else
    loop (i-1) (shuffle l)
in
let () = loop iters l in
let l = shuffle l in
let _ = List.fold_left
( fun (i, couple) name ->
  let couple = (List.rev (name::couple)) in
  if ((i>0) && i mod 2 <> 0) then
    let () = print_c couple in
    ((i+1), [])
  else
    ((i+1), couple)
) (0,[]) l
in
let count = List.length l in
if count mod 2 = 0 then
  Printf.printf " ....\n"
else
  let last_one = (List.hd (List.rev l)) in
  Printf.printf " %s....\n" last_one

File: santa.py
import sys
import random
from random import shuffle
a_string = sys.argv[1]
sum_ord=float(0.0)
for x in a_string:
  sum_ord = (ord(x)/100) + sum_ord
random.seed(sum_ord)
l = ["Eline VDV","Joris","Sebastiaan","Dirk","Kristof","Eline H","Pieter", "Petrus", "Eva"]
random.shuffle(l)
print("*****************************The super extrarandom kadoselector***************************")
first=l[0]
second=l[1]
second_last=l[-2]
last=l[-1]
zin=" koopt voor "
sys.stdout.write("%s%s%s\n" % (first, zin, second))
i=0
for x in xrange(1, (len(l)-1)):
  if i > 0:
    sys.stdout.write("%s%s%s\n" % (l[x-1], zin, l[x]))
    i = 0
  i = i + 1
sys.stdout.write("%s%s%s\n" % (second_last, zin, last))
sys.stdout.write("%s%s%s\n" % (last, zin, first))
print("***Het algoritme is niet verantwoordelijk voor het verkrijgen van een slecht kado*********")                                                                                                     
